src/mappings/mappingHandlers.ts

```diff
@@ -24,5 +29,49 @@ export async function handleMessage(message: CosmosMessage) {
         data: message.msg.decodedMsg,
     });
     await messageStore.save();
+
+    if(message.msg.typeUrl === "/ethermint.evm.v1.MsgEthereumTx") {
+        await handleEvmTransaction(message);
+    }
+}
+
+export async function handleEvmTransaction(message: CosmosMessage) {
+    const blockHeight = BigInt(message.block.block.header.height);
+    const tx = message.msg.decodedMsg as any;
+    const decodedTx = (global as any).registry.decodedMsg(tx.data);
```

Comment:
Maybe the types need updating but this should work
```suggestion
    const decodedTx = registry.decodedMsg(tx.data);
```